subprojects/packagefiles/joltphysics/meson.build

```
@@ -0,0 +1,684 @@
+  'Jolt/Core/STLAllocator.h',
+  'Jolt/Core/STLTempAllocator.h',
+  'Jolt/Core/JobSystemThreadPool.cpp',
+  'Jolt/Core/QuickSort.h',
+  'Jolt/Core/NonCopyable.h',
+  'Jolt/Math/Mat44.inl',
+  'Jolt/Math/DMat44.inl',
+  'Jolt/Math/DMat44.h',
+  'Jolt/Math/MathTypes.h',
+  'Jolt/Math/Double3.h',
+  'Jolt/Math/FindRoot.h',
+  'Jolt/Math/Mat44.h',
+  'Jolt/Math/Trigonometry.h',
+  'Jolt/Math/Swizzle.h',
+  'Jolt/Math/EigenValueSymmetric.h',
+  'Jolt/Math/Float3.h',
+  'Jolt/Math/BVec16.h',
+  'Jolt/Math/Float2.h',
+  'Jolt/Math/Vector.h',
+  'Jolt/Math/UVec4.h',
+  'Jolt/Math/Vec3.cpp',
+  'Jolt/Math/Vec3.inl',
+  'Jolt/Math/Quat.inl',
+  'Jolt/Math/Quat.h',
+  'Jolt/Math/Real.h',
+  'Jolt/Math/Vec3.h',
+  'Jolt/Math/Vec4.h',
+  'Jolt/Math/HalfFloat.h',
+  'Jolt/Math/GaussianElimination.h',
+  'Jolt/Math/Math.h',
+  'Jolt/Math/DVec3.inl',
+  'Jolt/Math/BVec16.inl',
+  'Jolt/Math/Vec4.inl',
+  'Jolt/Math/UVec4.inl',
+  'Jolt/Math/Float4.h',
+  'Jolt/Math/Matrix.h',
+  'Jolt/Math/DVec3.h',
+  'Jolt/Math/DynMatrix.h',
+  'Jolt/TriangleSplitter/TriangleSplitterMean.cpp',
+  'Jolt/TriangleSplitter/TriangleSplitter.h',
+  'Jolt/TriangleSplitter/TriangleSplitterBinning.cpp',
+  'Jolt/TriangleSplitter/TriangleSplitterBinning.h',
+  'Jolt/TriangleSplitter/TriangleSplitter.cpp',
+  'Jolt/TriangleSplitter/TriangleSplitterMean.h',
+]
+
+object_stream = get_option('object_stream')
+
+if object_stream
+  all += [
+    'Jolt/ObjectStream/ObjectStreamBinaryIn.h',
+    'Jolt/ObjectStream/ObjectStreamTextIn.cpp',
+    'Jolt/ObjectStream/SerializableAttribute.h',
+    'Jolt/ObjectStream/SerializableObject.h',
+    'Jolt/ObjectStream/ObjectStreamTextIn.h',
+    'Jolt/ObjectStream/TypeDeclarations.cpp',
+    'Jolt/ObjectStream/ObjectStream.h',
+    'Jolt/ObjectStream/ObjectStreamIn.cpp',
+    'Jolt/ObjectStream/ObjectStreamTextOut.cpp',
+    'Jolt/ObjectStream/ObjectStreamBinaryOut.cpp',
+    'Jolt/ObjectStream/ObjectStreamIn.h',
+    'Jolt/ObjectStream/GetPrimitiveTypeOfType.h',
+    'Jolt/ObjectStream/ObjectStreamBinaryIn.cpp',
+    'Jolt/ObjectStream/SerializableObject.cpp',
+    'Jolt/ObjectStream/ObjectStreamBinaryOut.h',
+    'Jolt/ObjectStream/ObjectStream.cpp',
+    'Jolt/ObjectStream/ObjectStreamOut.cpp',
+    'Jolt/ObjectStream/SerializableAttributeTyped.h',
+    'Jolt/ObjectStream/SerializableAttributeEnum.h',
+    'Jolt/ObjectStream/TypeDeclarations.h',
+    'Jolt/ObjectStream/ObjectStreamOut.h',
+    'Jolt/ObjectStream/ObjectStreamTextOut.h',
+    'Jolt/ObjectStream/ObjectStreamTypes.h',
+  ]
+endif
+
+headers = []
+sources = []
+
+foreach item : all
+  if item.endswith('.h') or item.endswith('.inl')
+    headers += item
+  elif item.endswith('.cpp')
+    sources += item
+  endif
+endforeach
+
+args = []
+build_args = []
+
+if get_option('default_library') == 'shared'
+  args += ['-DJPH_SHARED_LIBRARY']
+  build_args += ['-DJPH_BUILD_SHARED_LIBRARY']
+endif
+
+buildtype = get_option('buildtype')
+build_debug = buildtype.contains('debug')
```

This won't work on all Meson versions.

```suggestion
build_debug = get_option('debug')
```